src/python/pants/backend/go/goals/test.py

```diff
@@ -121,8 +139,36 @@ def transform_test_args(args: Sequence[str]) -> tuple[str, ...]:
 async def run_go_tests(
     field_set: GoTestFieldSet, test_subsystem: TestSubsystem, go_test_subsystem: GoTestSubsystem
 ) -> TestResult:
-    maybe_pkg_info = await Get(
-        FallibleFirstPartyPkgInfo, FirstPartyPkgInfoRequest(field_set.address)
+    maybe_pkg_info, wrapped_target, test_sources = await MultiGet(
+        Get(FallibleFirstPartyPkgInfo, FirstPartyPkgInfoRequest(field_set.address)),
+        Get(WrappedTarget, Address, field_set.address),
+        Get(SourceFiles, SourceFilesRequest([field_set.sources])),
+    )
+    target = wrapped_target.target
+    test_sources_dirpath = field_set.address.spec_path
+
+    explicit_dependencies = await Get(
+        ExplicitlyProvidedDependencies, DependenciesRequest(target[Dependencies])
+    )
+
+    # TODO: Consider ignores as well.
```

Comment:
How can `ignores` be applied to the `includes`?

Reply:
By using `DependenciesRequest`, which handles it for you.